### ISSUE TITLE
call order for `didInsertElement` for siblings reversed

### ISSUE THREAD

**Issue by KoKuToru** [emberjs/ember.js#13972]
Example:
```hbs
{{my-component name="A"}}
{{my-component name="B"}}
{{my-component name="C"}}
{{my-component name="D"}}
```
order of didInsertElement hooks: `A B C D`

in `2.9.0-alpha+8ba7f68f` the order is now: `D C B A`

Ember-Twiddle: https://ember-twiddle.com/#/847f6adf7c7dc53f65f449062c3c5549 (output in console)

I have some code that depends on the order of `didInsertElement`.
`willInsertElement` is still in the right order.

**Comment by rwjblue** [comment]
Hmm.  I don't think the order of `didInsertElement` for siblings was a really well defined API.  We definitely guarantee that any child components' `didInsertElement` is called before their parents, but I don't think we have made guarantees on the ordering of `A`, `B`, `C`, `D` in your example.

@krisselden - You likely know this area the best, what do you think?  Should we consider this a bug and try to fix it, or is the Glimmer@2 behavior intentional?

**Comment by chancancode** [comment]
I think we did it to pass tests: https://github.com/emberjs/ember.js/commit/158158364999302d1309b2135f7b46218496007f#diff-c8938eb5cb17b733982744e1621cb8e4R168

If the actual goal is to have the children called before the parent, I think the would be pretty simple; instead of queuing the hooks in `OpenComponentOpcode`, we could move it to `CloseComponentOpcode`, then we can delete the patch in Ember that reverses the order, and everything should just work.

Are the rules the same for update hooks? I don't know if there is an equivalent place for it now, but we can make `CloseComponentOpcode` emit a corresponding updating opcode to queue the hook.

**Comment by rwjblue** [comment]
@chancancode - The parent/child ordering is correct. This issue is about siblings.

**Comment by chancancode** [comment]
Yeah. Basically we did pre-order queueing (Parent - Child 1 - Child 2 - Child 3) and then the test failed, so we just reversed the order to make it pass (C3 C2 C1 P).

If we actually want post-order queuing (C1 C2 C3 P), I think we can actually do it. Might have some implications for exception cases though, since if it errored when appending one of your children, your hook will not be queued/called, but that whole area needs a lot more love anyway.
